dotCMS/src/integration-test/java/com/dotmarketing/business/web/UserWebAPIImplTest.java

```diff
@@ -0,0 +1,137 @@
+package com.dotmarketing.business.web;
+
+import static org.junit.Assert.assertTrue;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import com.dotcms.datagen.UserDataGen;
+import com.dotcms.mock.request.MockAttributeRequest;
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.WebKeys;
+
+public class UserWebAPIImplTest {
+
+
+    private static User frontEndUser, backEndUser;
+
+
+    private static UserWebAPIImpl userWebAPI;
+
+
+    private HttpServletRequest anonymousRequest() {
+        return new MockAttributeRequest(new MockHttpRequest("localhost", "/api/testing-web-resorce").request()).request();
+    }
+
+    private HttpServletRequest frontEndRequest() {
+        final HttpServletRequest request = anonymousRequest();
+        assertTrue("frontEndUser has frontEnd role", frontEndUser.isFrontendUser());
+        request.setAttribute(WebKeys.USER, frontEndUser);
+        return request;
+    }
+
+    private HttpServletRequest backEndRequest() {
+        final HttpServletRequest request = anonymousRequest();
+        assertTrue("backEndUser has backend role", backEndUser.isBackendUser());
+        request.setAttribute(WebKeys.USER, backEndUser);
+        return request;
+    }
+
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5106108)